utils/downloads.py

```diff
@@ -53,3 +54,29 @@ def download_sounds(licenses_url, pack):
     response = HttpResponse(filelist, content_type="text/plain")
     response['X-Archive-Files'] = 'zip'
     return response
+
+
+def should_suggest_donation(user, times_shown_in_last_day):
+    """
+    This method indicates when we should display the donation modal to the user. This will be based on 3 settings 
+    indicating how many days after a donation we show the modal again, after how many downloads we display the modal 
+    and for how long. The modal will be shown a maximum number of times per day.
+    """
+
+    if times_shown_in_last_day >= settings.DONATION_MODAL_DISPLAY_TIMES_DAY:
+        # If modal has been shown more than settings.DONATION_MODAL_DISPLAY_TIMES_DAY times, don't show it again today
+        return False
+
+    donation_period = datetime.datetime.now() - datetime.timedelta(days=settings.DONATION_MODAL_DAYS_AFTER_DONATION)
+    last_donation = user.donation_set.order_by('created').last()
+    if not last_donation or last_donation.created < donation_period:
+        # If there has never been a donation or last donation is older than settings.DONATION_MODAL_DAYS_AFTER_DONATION,
+        # check if the number of downloads in the last settings.DONATION_MODAL_DOWNLOAD_DAYS days if bigger than
+        # settings.DONATION_MODAL_DOWNLOADS_IN_PERIOD. If that is the case, show the modal.
+        num_downloads_in_period = Download.objects.filter(
```

Comment:
Fixed in https://github.com/MTG/freesound/pull/820/commits/00a2c207a695c5049c489d8cbccb5d4c766aa099